avogadro/core/crystaltools.h

```diff
@@ -102,6 +102,17 @@ class AVOGADROCORE_EXPORT CrystalTools
   static bool isNiggliReduced(const Molecule& mol);
 
   /**
+   * Build a supercell by expanding upon the unit cell of @a molecule. It will
+   * only return false if the molecule does not have a unit cell or if a, b, or
+   * c is set to zero.
+   * @param a The number of units along lattice vector a for the supercell
+   * @param b The number of units along lattice vector b for the supercell
+   * @param c The number of units along lattice vector c for the supercell
+   * @return True on success, false on failure.
+   */
+  static bool buildSupercell(Molecule &molecule, uint a, uint b, uint c);
```

Comment:
I don't think uint is part of C++, http://stackoverflow.com/questions/3552094/c-uint-unsigned-int-int seems to agree on a quick search. We should use unsigned int, and if others are lurking use the standard type. Qt provides some with size guarantees I think.